### PR TITLE
utils/Directory: add a way to set uid/gid

### PR DESCRIPTION
**Pull request by TomasTomecek**
Fixes: https://github.com/fedora-modularity/meta-test-family/issues/92

### CODE REVIEW

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: success, return code = __0__ 
* Result url: __[https://da.gd/Sw135](https://da.gd/Sw135)__
* Comment: __See details for logs__

File: tests/integration/test_utils.py [hunk 1]
```diff
@@ -70,6 +70,54 @@ def test_directory_acl():
     assert not os.path.isdir(str(d))
 
 
+def test_user_ownership_int():
```
Comment (dhodovsk) [inline tests/integration/test_utils.py]: These could be moved to unit tests.

File: conu/utils/filesystem.py [hunk 2]
```diff
@@ -49,8 +60,22 @@ def __init__(self, path, mode=None, facl_rules=None, selinux_context=None, selin
...
+                raise ConuException("User %r not found, error message: %r" % (user_owner, ex))
+        else:
+            self.owner = user_owner
+        if isinstance(group_owner, six.string_types):
+            try:
+                self.group = pwd.getpwnam(group_owner)[3]
+            except KeyError as ex:
+                raise ConuException("Group %r not found, error message: %r" % (group_owner, ex))
+        else:
+            self.group = group_owner
```
Comment (dhodovsk) [inline conu/utils/filesystem.py]: Beautiful! :)

File: conu/utils/filesystem.py [hunk 3]
```diff
@@ -39,6 +46,10 @@ def __init__(self, path, mode=None, facl_rules=None, selinux_context=None, selin
             if get_selinux_status() == "Disabled":
                 raise ConuException("You are trying to apply SELinux labels, but SELinux is "
                                     "disabled on this system. Please enable it first.")
+
+        # if set to True, it means the directory is created and set up
+        self._initialized = False
```
Comment (dhodovsk) [inline conu/utils/filesystem.py]: Attributes are being changed also when already existing directories are specified in path. Is this intended?

Comment (TomasTomecek) [reply conu/utils/filesystem.py]: From my PoV, yes. That's why I made sure that `os.makedirs` won't fail if the directory already exists.

Do you think this is bad?

File: conu/utils/filesystem.py (hunk 2 above)
Comment (TomasTomecek) [reply conu/utils/filesystem.py]: I added this after I wrote tests -- would be so annoying to people.

File: tests/integration/test_utils.py (hunk 1 above)
Comment (TomasTomecek) [reply tests/integration/test_utils.py]: This is what I wanna talk about. In my world, unit tests don't change anything -- they shouldn't create any files or change anything.

Comment (bkabrda) [reply tests/integration/test_utils.py]: So I'm not familiar with the wider context here, but I personally see no problem with creating temporary files during unit tests. What I'm missing here is a cleanup, though :)

Comment (TomasTomecek) [reply tests/integration/test_utils.py]: You freaked me out! Here's the cleanup: https://github.com/fedora-modularity/conu/pull/64/files#diff-fe8f58742ef88776b0c50edcb8d062bfR84

```
filesystem.py              104 INFO     initializing Directory(path=/tmp/qeruikoznj)
filesystem.py              115 INFO     initialized
filesystem.py               94 INFO     brace yourselves, removing '/tmp/qeruikoznj'
```

Comment (TomasTomecek) [reply tests/integration/test_utils.py]: moved to unit tests

Comment (bkabrda) [reply tests/integration/test_utils.py]: Ah, ok, didn't realize that the cleanup would be in the `__exit__` method.

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: success, return code = __0__ 
* Result url: __[https://da.gd/o1mA](https://da.gd/o1mA)__
* Comment: __See details for logs__

File: conu/utils/filesystem.py (hunk 3 above)
Comment (dhodovsk) [reply conu/utils/filesystem.py]: Yes, it extends usability of the class. I was just wondering if it wouldn't cause some problems, e.g when user unintentionally provides wrong path - relative path to some unrelated file. But apparently an edge-case, so approving :)

Comment (TomasTomecek) [reply conu/utils/filesystem.py]: so I really hope that no one will do path=/

**Pull request merged**